### PR TITLE
Add tracks events

### PR DESCRIPTION
**Pull request by oikeme**
Add tracks events for payment method improvements for Jetpack Manage

Closes https://github.com/Automattic/jetpack-genesis/issues/171

## Proposed Changes
Adds tracks events for updates to adding new payments in Jetpack Manage

Events include
* The `Add New Card` button is clicked
* The `Set as primary payment methods` toggle is clicked when adding a new payment
* The `Submit` button is clicked to add a new card
* The 3 dot menu below a payment card in the payment methods page is clicked
* The `Set as primary card` option in the card actions menu is clicked
* The delete option is clicked in the card actions menu

## Testing Instructions

* Use the Calypso Live link below to access Jetpack Manage > Purchases >Payment Methods page and append `?flags=jetpack/card-addition-improvements` to the url.
* Open dev tools and the Network tab. Filter all `*.gif` requests. 
* Ensure that the following events are triggered for the actions
* Clicking the Add New Card button - `calypso_partner_portal_payments_add_new_card_button_click`
Screenshot 2024-01-16 at 7 58 18 PM
* Toggle the `Set as primary payment method` checkbox in the page to add a new payment method - calypso_partner_portal_add_credit_card_as_default_toggle
Screenshot 2024-01-16 at 7 58 52 PM
* Clicking the Submit button to add a new payment method - `calypso_partner_portal_add_credit_card_form_submit`
Screenshot 2024-01-16 at 7 58 52 PM
* Clicking the 3 dot menu on a card in the payments method page - `calypso_partner_portal_payments_card_actions_button_click`
Screenshot 2024-01-16 at 7 58 32 PM
* Clicking the Set as primary option in the card actions menu - `calypso_partner_portal_payments_card_actions_set_as_primary_click`
Screenshot 2024-01-16 at 7 58 40 PM
* Clicking the Delete option in the card actions menu - `calypso_partner_portal_payments_card_actions_delete_click`
Screenshot 2024-01-16 at 8 09 18 PM

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/jetpack-cloud/sections/partner-portal/stored-credit-card-v2/credit-card-actions.tsx [hunk 1]
```diff
@@ -1,3 +1,4 @@
+import { recordTracksEvent } from '@automattic/calypso-analytics';
```
Comment (yashwin) [inline client/jetpack-cloud/sections/partner-portal/stored-credit-card-v2/credit-card-actions.tsx]: It is recommended to always import like below and use the Middleware:

```
import { recordTracksEvent } from 'calypso/state/analytics/actions';
```

Note: [Unless you have a strong reason to call `recordTracksEvent` directly, you should use the Analytics Middleware instead](https://github.com/Automattic/wp-calypso/blob/2544768a56d3a10a77cacf8c1bff65ab78fb6881/packages/calypso-analytics/README.md)

File: client/jetpack-cloud/sections/partner-portal/stored-credit-card-v2/index.tsx [hunk 2]
```diff
@@ -2,11 +2,11 @@ import { PaymentLogo } from '@automattic/wpcom-checkout';
 import classNames from 'classnames';
 import { useTranslate } from 'i18n-calypso';
 import PaymentMethodDeleteDialog from 'calypso/jetpack-cloud/sections/partner-portal/payment-method-delete-dialog';
+import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
```
Comment (yashwin) [inline client/jetpack-cloud/sections/partner-portal/stored-credit-card-v2/index.tsx]: Same as above, let's use the Middleware

**Review by yashwin — CHANGES_REQUESTED** [review]
Thanks for working on this, @oikeme.

Everything works as expected. 

I have left a few comments.

Also, looks there there are conflicts.

File: client/jetpack-cloud/sections/partner-portal/stored-credit-card-v2/credit-card-actions.tsx [hunk 3]
```diff
@@ -21,6 +22,7 @@ export default function CreditCardActions( {
 
 	const showActions = () => {
 		setIsOpen( true );
+		recordTracksEvent( 'calypso_partner_portal_payments_card_actions_button_click' );
```
Comment (yashwin) [inline client/jetpack-cloud/sections/partner-portal/stored-credit-card-v2/credit-card-actions.tsx]: We will have to dispatch this event-tracking call.

```
const dispatch = useDispatch();

dispatch( recordTracksEvent('calypso_partner_portal_payments_card_actions_button_click');
```

File: client/jetpack-cloud/sections/partner-portal/stored-credit-card-v2/index.tsx [hunk 4]
```diff
@@ -50,13 +50,15 @@ export default function StoredCreditCardV2( {
...
 				} );
+				recordTracksEvent( 'calypso_partner_portal_payments_card_actions_set_as_primary_click' );
 			},
 		},
 		{
 			name: translate( 'Delete' ),
 			isEnabled: true,
 			onClick: () => {
 				setIsDeleteDialogVisible( true );
+				recordTracksEvent( 'calypso_partner_portal_payments_card_actions_delete_click' );
```
Comment (yashwin) [inline client/jetpack-cloud/sections/partner-portal/stored-credit-card-v2/index.tsx]: Same as above. We will have to dispatch these event-tracking calls.

**Review by yashwin — CHANGES_REQUESTED** [review]
Thanks for addressing my comments, @oikeme!

I have left a few comments.

File: client/jetpack-cloud/sections/partner-portal/stored-credit-card-v2/index.tsx (hunk 4 above)
Comment (oikeme) [reply client/jetpack-cloud/sections/partner-portal/stored-credit-card-v2/index.tsx]: I've updated both and thanks!

**Review by yashwin — DISMISSED** [review]
Thanks for addressing my comments. 

Looks good me to now :shipit:

File: client/jetpack-cloud/sections/partner-portal/stored-credit-card-v2/index.tsx [hunk 5]
```diff
@@ -50,13 +52,17 @@ export default function StoredCreditCardV2( {
...
+					recordTracksEvent( 'calypso_partner_portal_payments_card_actions_set_as_primary_click' )
+				);
 			},
 		},
 		{
 			name: translate( 'Delete' ),
 			isEnabled: true,
 			onClick: () => {
 				setIsDeleteDialogVisible( true );
+				recordTracksEvent( 'calypso_partner_portal_payments_card_actions_delete_click' );
```
Comment (yashwin) [inline client/jetpack-cloud/sections/partner-portal/stored-credit-card-v2/index.tsx:65]:
```suggestion
				dispatch( recordTracksEvent( 'calypso_partner_portal_payments_card_actions_delete_click' )
				);
```

**Review by yashwin — CHANGES_REQUESTED** [review]
Sorry for missing this earlier. I have left one comment.

File: client/jetpack-cloud/sections/partner-portal/stored-credit-card-v2/index.tsx (hunk 5 above)
Comment (yashwin) [inline client/jetpack-cloud/sections/partner-portal/stored-credit-card-v2/index.tsx]: We need to dispatch this event.

**Review by yashwin — APPROVED** [review]
Looks good now :shipit:

**Pull request merged**